app/serializers/v2/person_serializer.rb

```diff
@@ -19,10 +19,13 @@ class PersonSerializer
     has_one :ethnicity, serializer: EthnicitySerializer
     has_one :gender, serializer: GenderSerializer
 
+    # NB: we need to lazy load the category to prevent an unnecessary call to Nomis
+    has_one :category, serializer: CategorySerializer, id_method_name: :key, lazy_load_data: true, &:category
```

Comment:
Not sure this needs to have any special treatment - ideally we should persist the category against the profile when the Nomis API call completes. I'd prefer that the lookup happens explicitly in the people controller and the category relationship is a standard `belongs_to :category` (or maybe `has_one` - whichever you feel is more semantically correct 🤔 ) in the `Person` model.

Also this file could be simplified using inheritance as per @cwrw suggestion...

```ruby
module V2
  class PersonSerializer < PeopleSerializer
     has_one :category, serializer: CategorySerializer

    SUPPORTED_RELATIONSHIPS = %w[ethnicity gender profiles category].freeze
  end
end
```

Sorry to say this but I think it would also be good to migrate the existing `category` attribute on the `Person` model to the corresponding `category_id`. This moves closer to an idiomatic approach which is generally preferred.

Reply:
Discussed this offline, the `has_one` implementation of `category` you have here is fine as is, but would still be nice to introduce inheritance for a cleaner serializer.

Reply:
The person serializers now have inheritance: `class PersonWithCategorySerializer < ::V2::PersonSerializer`